### ISSUE TITLE
Websocket Closed on chrome but no on safari (proxy behind)

### ISSUE THREAD

**Issue by barciella** [jupyter/notebook#2803]
When I open a notebook on chrome the logs are:

`[I 18:35:02.082 NotebookApp] Adapting to protocol v5.1 for kernel 28d14fb3-8ded-4024-b36a-598778c0bc1a
[D 18:35:02.083 NotebookApp] 101 GET /api/kernels/28d14fb3-8ded-4024-b36a-598778c0bc1a/channels?session_id=C3A7DF49C6E842D7A8A56A6F4958D9E5 (127.0.0.1) 4.41ms
[D 18:35:02.084 NotebookApp] Opening websocket /api/kernels/28d14fb3-8ded-4024-b36a-598778c0bc1a/channels
[D 18:35:02.084 NotebookApp] Connecting to: tcp://127.0.0.1:40835
[D 18:35:02.085 NotebookApp] Connecting to: tcp://127.0.0.1:52911
[D 18:35:02.086 NotebookApp] Connecting to: tcp://127.0.0.1:40293
[D 18:35:02.239 NotebookApp] activity on 28d14fb3-8ded-4024-b36a-598778c0bc1a: status
[D 18:35:02.242 NotebookApp] activity on 28d14fb3-8ded-4024-b36a-598778c0bc1a: status
[D 18:35:02.388 NotebookApp] Websocket closed 28d14fb3-8ded-4024-b36a-598778c0bc1a:C3A7DF49C6E842D7A8A56A6F4958D9E5
[D 18:35:04.257 NotebookApp] Initializing websocket connection /api/kernels/28d14fb3-8ded-4024-b36a-598778c0bc1a/channels
[I 18:35:04.260 NotebookApp] Adapting to protocol v5.1 for kernel 28d14fb3-8ded-4024-b36a-598778c0bc1a
[D 18:35:04.261 NotebookApp] 101 GET /api/kernels/28d14fb3-8ded-4024-b36a-598778c0bc1a/channels?session_id=C3A7DF49C6E842D7A8A56A6F4958D9E5 (127.0.0.1) 4.45ms
[D 18:35:04.261 NotebookApp] Opening websocket /api/kernels/28d14fb3-8ded-4024-b36a-598778c0bc1a/channels
[D 18:35:04.261 NotebookApp] Connecting to: tcp://127.0.0.1:40835
[D 18:35:04.262 NotebookApp] Connecting to: tcp://127.0.0.1:52911
[D 18:35:04.263 NotebookApp] Connecting to: tcp://127.0.0.1:40293
[D 18:35:04.418 NotebookApp] activity on 28d14fb3-8ded-4024-b36a-598778c0bc1a: status
[D 18:35:04.421 NotebookApp] activity on 28d14fb3-8ded-4024-b36a-598778c0bc1a: status
[D 18:35:04.564 NotebookApp] Websocket closed 28d14fb3-8ded-4024-b36a-598778c0bc1a:C3A7DF49C6E842D7A8A56A6F4958D9E5
`

But when I connect with safari everything works. 

I am behind a proxy and I have no idea why it is working on safari but not on chrome. I tried on incognito mode as well.

**Comment by takluyver** [comment]
Check in your proxy settings that 127.0.0.1 is an exception that it shouldn't use the proxy for.

**Comment by barciella** [comment]
Tried that and had the same problem. Chrome and Firefox doesn't work but Safari does.

**Comment by takluyver** [comment]
Firefox can be configured to use system proxy settings or its own proxy settings - check in there to see what it's doing.

**Comment by barciella** [comment]
I am on MacOS Sierra btw.
From chrome, inspecting the WebSocket I have infinites 101 status.
From safari everything works.
From Firefox: WebSocket connection failed:. 

What I mean about the proxy behind is that I have an oauth2 implemented to connect to the notebook. I was running a very old version of all packages and it worked fine. I updated everything and now only Safari can connect on the notebook. I am stumped of what Safari does different and which package is the difference.

**Comment by takluyver** [comment]
I don't understand what you mean about oauth2. Are you using Jupyterhub? Or have you got some other authentication system in front of it?

**Comment by barciella** [comment]
Yes I am using an authentication system in front. https://github.com/bitly/oauth2_proxy

It is probably some configuration on it but I can't debug exactly what. I want to understand how I can do that.

**Comment by takluyver** [comment]
No idea, sorry, I've never come across that package before.

If you want to put notebooks behind oauth, Jupyterhub is how we'd recommend doing it. It might be overkill if it's for a single user, though.
